src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerTypeSymbol.cs

```diff
@@ -19,13 +19,20 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
     internal sealed partial class FunctionPointerTypeSymbol : TypeSymbol
     {
         public static FunctionPointerTypeSymbol CreateFromSource(FunctionPointerTypeSyntax syntax, Binder typeBinder, DiagnosticBag diagnostics, ConsList<TypeSymbol> basesBeingResolved, bool suppressUseSiteDiagnostics)
-            => new FunctionPointerTypeSymbol(
-                FunctionPointerMethodSymbol.CreateFromSource(
-                    syntax,
-                    typeBinder,
-                    diagnostics,
-                    basesBeingResolved,
-                    suppressUseSiteDiagnostics));
+        {
+            if (!typeBinder.InUnsafeRegion)
+            {
+                diagnostics.Add(ErrorCode.ERR_UnsafeNeeded, syntax.Location);
```

Comment:
>syntax.Location [](start = 60, length = 15)

Consider reducing the span for the error. Perhaps to the ```delegate*```.

Reply:
Why? Personally, I'd prefer it to be on the whole thing.

---
In reply to: [376031475](https://github.com/dotnet/roslyn/pull/41455#discussion_r376031475) [](ancestors = 376031475)

Reply:
> Why? Personally, I'd prefer it to be on the whole thing.

What is the benefit of squiggling the whole signature, which, given the nature of the type, could span multiple lines? The benefit of reducing the span is to avoid intersection with other errors in the type, which makes it easier to locate their source.

---
In reply to: [376094583](https://github.com/dotnet/roslyn/pull/41455#discussion_r376094583) [](ancestors = 376094583,376031475)